Launcher/MainWindow.xaml.cs

```diff
@@ -210,15 +219,64 @@ void patch_exes_for_large_address_support()
             }
         }
 
+        void repair_registry()
+        {
+            string H2Tool_Path = AppDomain.CurrentDomain.BaseDirectory;
+
+            if (Registry.GetValue(H2EK_key, Tools_Install_Directory, null) is null || Registry.GetValue(Guerilla_key, Tools_Directory, null) is null || Force_Registry_Fix is true)
+            {
+                RegistryKey H2EK_Install_Path_key = RegistryKey
+                    .OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry32)
+                    .CreateSubKey("SOFTWARE\\Microsoft\\Microsoft Games\\Halo 2\\1.0", true);
+
+                RegistryKey Guerilla_Tag_key = RegistryKey
+                    .OpenBaseKey(RegistryHive.LocalMachine, RegistryView.Registry32)
+                    .CreateSubKey("SOFTWARE\\Microsoft\\Halo 2", true);
+
+                if (Force_Registry_Fix is true)
```

Comment:
```suggestion
                if (Force_Registry_Fix == true)
```
That would be the more conventional way to do that afaik